autoblocks/_impl/tracer.py

```diff
@@ -101,6 +109,52 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    async def _evaluate_event(self, event: TracerEvent, evaluator: BaseEventEvaluator) -> Optional[EventEvaluation]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                log.error("Event evaluation threw an exception", err)
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                log.error("Event evaluation threw an exception", err)
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: BaseEventEvaluator, event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[EventEvaluation]:
+        if len(evaluators) == 0:
+            return []
+        try:
+            evaluations: List[EventEvaluation] = await gather_with_max_concurrency(
```

Comment:
is this type hint correct? the filter below suggests this isn't a list of EventEvaluation